src/formatDate.js

```diff
@@ -8,7 +8,39 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+
+  const newFormat = fromFormat.slice(0, -1);
+  const newForm = toFormat.slice(0, -1);
+
+  const parts = date.split(fromSeparator);
+
+  const dateObject = {};
+
+  for (let i = 0; i < newFormat.length; i++) {
+    dateObject[newFormat[i]] = parts[i];
+  }
+
+  if (newFormat.includes('YYYY') && newForm.includes('YY')) {
+    dateObject['YY'] = dateObject['YYYY'].slice(-2);
+  } else if (newFormat.includes('YY') && newForm.includes('YYYY')) {
+    const year = +dateObject['YY'];
+
+    dateObject['YYYY'] =
+      year >= 30 ? `19${dateObject['YY']}` : `20${dateObject['YY']}`;
+  }
```

Comment:
The logic for converting between 'YYYY' and 'YY' formats assumes that years '30' and above belong to the 1900s, while years below '30' belong to the 2000s. This assumption might not be valid for all use cases. Consider making this logic configurable or document this assumption clearly.